proj/src/main/webapp/chat.js

```diff
@@ -501,11 +504,62 @@ function createMessageWithTemplate(key, messageObj) {
     timestamp.innerText = new Date(messageObj.timestamp).toLocaleString();
 
     message.id = key;
+    message.dataset.userId = messageObj.senderUID;
     addUsernameToMessage(messageObj.senderUID, message)
     return message;
 }
 
+// onclick for messages
+function loadProfileOfSender(domElement, uid) {
+    domElement.addEventListener('click', function() {
+        populateProfileSidebar(uid);
+    })
+}
 
+function friendRequestButton(uid) {
+    const currUserRef = firebase.database().ref('/users/'+currentUID+'/friend-requests/'+uid);
+    const otherUserRef = firebase.database().ref('/users/'+uid+'/friend-requests/'+currentUID);
+
+    const buttonHouse = document.getElementById('button-house');
+    const button = document.createElement('button');
```

Comment:
Why is this button being created here? Can you create it in the html and just display or hide it conditionally?

Reply:
fixed